### ISSUE TITLE
Proposal: gen deepcopy code for each model

### ISSUE THREAD

**Issue by zhuangqh** [go-swagger/go-swagger#2129]
## Motivation

In some usecases, we expect to do deepcopy for api types. It would be more efficient to generate deepcopy code for swagger model.

## Proposal

Generate `DeepCopy() $object` method for each swagger model. We can take reference to https://github.com/kubernetes/code-generator.

**Comment by fredbi** [comment]
Yeah i ve seen that one too. We already have this for strfmt types. A PR with that addition would be welcome and I don t think it is too hard.. could be an additional option in model generation.

I would be happy to provide with any guidance you might need to implement this on our slack channel.

Cheers

Fred

**Comment by Swathi-L** [comment]
Hi @fredbi / @zhuangqh, is this enhancement completed?

**Comment by casualjim** [comment]
Feel free to work on it, it has not been completed

**Comment by Yuto2000** [comment]
@fredbi 
I am interested in this contrivance.
Please invite me to that slasck channel. Or let me take the guidance.
I would be happy to help!!

**Comment by houshengbo** [comment]
Hi folks, what the status of this feature. I am wondering if we support it now or not in swagger.

**Comment by usmanovbf** [comment]
Hi there! is it still actual?
